server/ctools_direct_response.rb

```diff
@@ -65,17 +69,38 @@ def extractAssignment(assignment)
 
     # Allow string replacement.  This is needed to ensure that host names are correct.
     # See studentdashboard.yml.TXT for information.
-    @stringReplace.each_pair do |key,value|
+    @stringReplace.each_pair do |key, value|
       next if assign[key.to_sym].nil?
       from_name = @stringReplace[key][0]
       to_name = @stringReplace[key][1]
-      assign[key.to_sym].gsub!(from_name,to_name)
+      assign[key.to_sym].gsub!(from_name, to_name)
       logger.debug "#{__FILE__}: #{__LINE__}: possible update for #{assign[key.to_sym]}"
     end
 
     assign
   end
 
+  # Take the raw assignment value and check to see if it should be filtered out.  Return the
+  # assignment if acceptable and nil otherwise.
+  # only accept assignments with calendar item label key of assignment.due.date
+  #"calendarItem": {
+  #"calendarTimeLabelKey": "assignment.due.date"
+
+  # Make static method since it requires only local data and that's easier for testing.
+  def self.filter(assignment)
+
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: assignment: [#{assignment.inspect}]"
+    calendarItem = assignment['calendarItem']
+    return nil if calendarItem.nil?
+
+    calendarTimeLabelKey = calendarItem['calendarTimeLabelKey']
+    return nil if calendarTimeLabelKey.nil?
```

Comment:
Is it necessary to test calendarTimeLabelKey for nil *and* equality to "assignment.due.date"?  Is there a test to check for exactly "assignment.due.date" and return false if it's not that value or it's nil?

If there's not a convenient single test, do both of those tests need to be in separate conditional statements?  Couldn't they could be combined?